### PR TITLE
Test descriptions

### PR DESCRIPTION
**Pull request by N-Wouda**
This PR adds a bundle of docstrings to our tests, removes duplicate tests, improves existing ones, and adds a few more functional tests in places that had open TODOs.

<details>

**Notes**:

- It is essential that you add a test when making code changes.
  This keeps the code coverage level up, and helps ensure the changes work as intended.
- When fixing a bug, you must add a test that would produce the bug in the master branch, and then show that it is fixed with the new code. 
- New code additions must be well formatted. Changes should pass the pre-commit workflow, which you can set up locally using [pre-commit](https://pre-commit.com/#intro). 
- Docstring additions must render correctly, including escapes and LaTeX.

</details>

### CODE REVIEW

File: pyvrp/Model.py [hunk 1]
```diff
@@ -181,6 +181,13 @@ def add_edge(
         if distance < 0 or duration < 0:
             raise ValueError("Cannot have negative edge distance or duration.")
 
+        if max(distance, duration) > MAX_USER_VALUE:
```
Comment (N-Wouda) [inline pyvrp/Model.py:184]: I moved this to `Model.add_edge`, which seemed like a more logical place for it.

File: pyvrp/search/tests/test_LocalSearch.py [hunk 2]
```diff
@@ -21,20 +21,13 @@
 from pyvrp.tests.helpers import make_heterogeneous, read
 
 
-def test_local_search_returns_same_solution_when_there_are_no_operators():
```
Comment (N-Wouda) [inline pyvrp/search/tests/test_LocalSearch.py:24]: This test duplicates a test further below. I noticed that while writing the same docstring twice :)

File: pyvrp/tests/test_Solution.py [hunk 3]
```diff
@@ -17,6 +17,9 @@
 
 
 def test_route_constructor_raises_for_empty_routes():
+    """
+    Tests that construction a ``Solution`` with empty routes fails.
```
Comment (leonlan) [inline pyvrp/tests/test_Solution.py:21]:
```suggestion
    Tests that constructing a ``Solution`` with empty routes fails.
```

File: pyvrp/tests/test_TimeWindowSegment.py [hunk 4]
```diff
@@ -52,5 +62,45 @@ def test_merge_multiple():
...
+    assert_(merged12.total_time_warp() != merged21.total_time_warp())
+
+    # Both segments start with 1 initial time warp. Going from depot -> client
+    # happens at t = 5 - 1, and takes 4 time units to complete. So we arrive at
+    # t = 8, which is three units after the time window closes. This adds 2
+    # time warp to the segment, which, together with the initial time warps
+    # makes for 2 + 1 + 1 = 4 total time warp.
+    assert_equal(merged12.total_time_warp(), 4)
+
+    # We can leave at the earliest as t = 3, the start of the client's time
```
Comment (leonlan) [inline pyvrp/tests/test_TimeWindowSegment.py:94]:
```suggestion
    # We can leave at the earliest at t = 3, the start of the client's time
```

**Review by leonlan — APPROVED** [review]

**Pull request merged**